protobuff-support/src/main/java/io/micronaut/protobuf/convert/ByteBufToProtoMessageConverter.java

```diff
@@ -56,10 +59,10 @@ public ByteBufToProtoMessageConverter(ProtobufferCodec codec) {
 
     private Optional<Message> rehydrate(ByteBuf object, Message.Builder builder) {
         try {
-            builder.mergeFrom(new ByteBufInputStream(object), codec.getExtensionRegistry());
+            builder.mergeFrom(new ByteBufInputStream(object.copy()), codec.getExtensionRegistry());
```

Comment:
I believe if you are making a copy here you will need to call `release()` on the byte buf to ensure a memory leak is not introduced

Reply:
I think we can use `public ByteBufInputStream(ByteBuf buffer, boolean releaseOnClose)` here to avoid memory leak and explicit `.close()` call